src/reference/asciidoc/overview.adoc

```diff
@@ -273,6 +273,106 @@ If you need to send a messages during startup, implement `ApplicationListener` a
 Alternatively, implement `SmartLifecycle`, put your bean in a late phase, and send the messages from the `start()` method.
 
 
+[[shaded]]
+=== Considerations When using Packaged (e.g. Shaded) Jars
+
+Spring Integration bootstraps certain features using Spring Framework's `SpringFactories` mechanism to load several `IntegrationConfigurationInitializer` classes.
+This includes the `-core` jar as well as certain others such as `-http`, `-jmx`, etc.
+The information for this process is stored in a file `META-INF/spring.factories` in each jar.
+
+Some developers prefer to repackage their application and all dependencies into a single jar using well-known tools, such as the https://maven.apache.org/plugins/maven-shade-plugin/[Apache Maven Shade Plugin].
+
+At the time of writing, the shade plugin does not merge the `spring.factories` files; you just get one which appears to be the first one encountered by the plugin when it extracts the jar contents.
```

Comment:
I still worry here about all other not merged `META-INF` entities. And, of course, about all other Spring projects. In most cases we develop target apps not only with Spring Integration, so such a shading is much broader issue than just Spring Integration's `spring.factories`.

See my comment in the related JIRA: https://jira.spring.io/browse/INT-4400?focusedCommentId=157056&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-157056

Reply:
I agree - let's worry about the others if/when someone requests it.